src/YahooFinance.py

```diff
@@ -6,103 +6,103 @@
 
 
 class YahooFinanceQuote:
-  # Expects the ticker symbol as the only argument.
-  # This can theoretically request multiple comma-separated symbols.
-  # This could theoretically be trimmed down by using `fields=` parameter.
-  URL_TEMPLATE = 'https://query1.finance.yahoo.com/v7/finance/quote?symbols={}'
-
-  @classmethod
-  def _construct_url(cls, ticker_symbol):
-    return YahooFinanceQuote.URL_TEMPLATE.format(ticker_symbol)
-
-  def __init__(self, ticker_symbol):
-    self.ticker_symbol = ticker_symbol.replace('.', '-')
-    self.url = YahooFinanceQuote._construct_url(self.ticker_symbol)
-    self.current_price = None
-    self.market_cap = None
-    self.name = None
-    self.average_volume = None
-    self.ttm_eps = None
-
-  def parse_quote(self, content):
-    data = json.loads(content)
-    results = data.get('quoteResponse', {}).get('result', [])
-    if not results:
-      return False
-    success = self._parse_current_price(results)
-    success = success and self._parse_market_cap(results)
-    success = success and self._parse_name(results)
-    success = success and self._parse_average_volume(results)
-    success = success and self._parse_ttm_eps(results)
-    return success
-
-  def _parse_current_price(self, results):
-    if results:
-      self.current_price = results[0].get('regularMarketPrice', None)
-    return True if self.current_price else False
-
-  def _parse_market_cap(self, results):
-    if results:
-      self.market_cap = results[0].get('marketCap', None)
-    return True if self.market_cap else False
-
-  def _parse_name(self, results):
-    if results:
-      self.name = results[0].get('longName', None)
-    return True if self.name else False
-
-  def _parse_average_volume(self, results):
-    if results:
-      regularMarketVolume = results[0].get('regularMarketVolume', -1)
-      averageDailyVolume3Month = results[0].get('averageDailyVolume3Month', -1)
-      averageDailyVolume10Day = results[0].get('averageDailyVolume10Day', -1)
-      self.average_volume = max(0, min(regularMarketVolume, averageDailyVolume3Month, averageDailyVolume10Day))
-    return True if self.average_volume else False
-    
-  def _parse_ttm_eps(self, results):
-    if results:
-      self.ttm_eps = results[0].get('epsTrailingTwelveMonths', None)
-    return True if self.ttm_eps else False
+    # Expects the ticker symbol as the only argument.
+    # This can theoretically request multiple comma-separated symbols.
+    # This could theoretically be trimmed down by using `fields=` parameter.
+    URL_TEMPLATE = 'https://query1.finance.yahoo.com/v6/finance/quote?symbols={}'
```

Comment:
This appears to have changed from v6 to v7. Was that part of this fix? (TBH idk the differences. I likely grabbed the v7 URL from some network request while trying to reverse engineer how the page is loaded)

Reply:
`v7` was already used in code. And as I checked, thic have caused `401` error returned. This was not logged anywhere ( now is ) and caused error from issue. Changing version to 6 fixed problem. Like, It is not undefined anymore. So I gguess it have fixed it.

Reply:
Sounds good to me.